File: ant-sdk/antd-js/examples/02-data.ts
/**
 * Example 02: Store and retrieve public data, with cost estimation.
 *
 * Prerequisite: antd daemon running on local testnet.
 */

import { createClient } from "../src/index.js";

const client = createClient();

// Estimate cost before storing
const payload = Buffer.from("Hello, Autonomi network!");
const est = await client.dataCost(payload);
console.log(
  `Estimate: ${est.fileSize} bytes in ${est.chunkCount} chunks, ` +
    `storage ${est.cost} atto, gas ${est.estimatedGasCostWei} wei, ` +
    `mode ${est.paymentMode}`
);

// Store public data
const result = await client.dataPutPublic(payload);
console.log(`Stored at address: ${result.address}`);
console.log(`Chunks stored: ${result.chunksStored}, payment mode: ${result.paymentModeUsed}`);

// Retrieve it back
const data = await client.dataGetPublic(result.address);
console.log(`Retrieved: ${data.toString()}`);

if (!data.equals(payload)) throw new Error("Round-trip mismatch!");
console.log("Public data round-trip OK!");

File: ant-sdk/antd-py/examples/02_data.py
"""Example 02: Store and retrieve public data, with cost estimation.

Prerequisite: antd daemon running on local testnet.
"""

from antd import AntdClient

client = AntdClient()

# Estimate cost before storing
payload = b"Hello, Autonomi network!"
est = client.data_cost(payload)
print(
    f"Estimate: {est.file_size} bytes in {est.chunk_count} chunks, "
    f"storage {est.cost} atto, gas {est.estimated_gas_cost_wei} wei, "
    f"mode {est.payment_mode}"
)

# Store public data
result = client.data_put_public(payload)
print(f"Stored at address: {result.address}")
print(f"Chunks stored: {result.chunks_stored}, payment mode: {result.payment_mode_used}")

# Retrieve it back
data = client.data_get_public(result.address)
print(f"Retrieved: {data.decode()}")

assert data == payload, "Round-trip mismatch!"
print("Public data round-trip OK!")
